### PR TITLE
chart: set overide_host_root as an env var rather than a config entry

### PR DESCRIPTION
**Pull request by roobre**
This is needed as env vars take precedence over config values. We were setting this as a config value while unaware that the image we ship set these env vars in the Dockerfile.

### CODE REVIEW

File: charts/newrelic-infrastructure/tests/unprivileged_override_host_root_test.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      licenseKey: test
+      cluster: test
+      privileged: false
+    asserts:
+      - contains:
+          path: spec.template.spec.containers[1].env
+          content:
+            name: "NRIA_OVERRIDE_HOST_ROOT"
+            value: ""
+        template: templates/kubelet/daemonset.yaml
```
Comment (kang-makes) [inline charts/newrelic-infrastructure/tests/unprivileged_override_host_root_test.yaml:33]: Could you test `notContains` when privileged is set to false?

**Review by kang-makes — APPROVED** [review]
It was a good idea when I thought about it but I didn't think about the env priority. 

I left a NIT about testing but LGMT.

Thanks for taking care of this.

File: charts/newrelic-infrastructure/tests/unprivileged_override_host_root_test.yaml (hunk 1 above)
Comment (roobre) [reply charts/newrelic-infrastructure/tests/unprivileged_override_host_root_test.yaml:33]: ... That was what this test was supposed to test. Now I wonder how it's passing.

Comment (roobre) [reply charts/newrelic-infrastructure/tests/unprivileged_override_host_root_test.yaml:33]: WDYM exactly? The env var _should_ be there (set to `""`) when `privileged: false`.

**Comment by roobre** [comment]
Made a quick manual test and this indeed seems to work as expected, no extraneous errors in unprivileged mode.

**Pull request merged**